src/mds/MDCache.cc

```diff
@@ -3083,6 +3083,7 @@ void MDCache::handle_resolve(MMDSResolve *m)
 
   if (mds->get_state() < MDSMap::STATE_RESOLVE) {
     if (mds->get_want_state() == CEPH_MDS_STATE_RESOLVE) {
+      discard_delayed_resolve(from);
```

Comment:
Not very sure about this too, but since the caller can be `process_delayed_resolve `, it can happen theoretically without considering any further:-(